localstack/services/s3/resource_providers/aws_s3_bucket.py

```diff
@@ -0,0 +1,722 @@
+        model["RegionalDomainName"] = domain_name
+        # by default (parity) s3 website only supports http
+        #   https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteHosting.html
+        #   "Amazon S3 website endpoints do not support HTTPS. If you want to use HTTPS,
+        #   you can use Amazon CloudFront [...]"
+        model[
+            "WebsiteURL"
+        ] = f"http://{model['BucketName']}.{S3_STATIC_WEBSITE_HOSTNAME}:{get_edge_port_http()}"
+        # resource["Properties"]["DualStackDomainName"] = ?
+
+    def _create_bucket_if_does_not_exist(self, model, region_name, s3_client):
+        try:
+            s3_client.head_bucket(Bucket=model["BucketName"])
+        except s3_client.exceptions.ClientError as e:
+            if e.response["Error"]["Message"] != "Not Found":
+                return
+
+            params = {
+                "Bucket": model["BucketName"],
+                "ACL": self._convert_acl_cf_to_s3(model.get("AccessControl", "PublicRead")),
+            }
+
+            if region_name != "us-east-1":
+                params["CreateBucketConfiguration"] = {
+                    "LocationConstraint": region_name,
+                }
+
+            s3_client.create_bucket(**params)
+
+    def _convert_acl_cf_to_s3(self, acl):
+        """Convert a CloudFormation ACL string (e.g., 'PublicRead') to an S3 ACL string (e.g., 'public-read')"""
+        return re.sub("(?<!^)(?=[A-Z])", "-", acl).lower()
+
+    def read(
+        self,
+        request: ResourceRequest[S3BucketProperties],
+    ) -> ProgressEvent[S3BucketProperties]:
+        """
+        Fetch resource information
+
+        IAM permissions required:
+          - s3:GetAccelerateConfiguration
+          - s3:GetLifecycleConfiguration
+          - s3:GetBucketPublicAccessBlock
+          - s3:GetAnalyticsConfiguration
+          - s3:GetBucketCORS
+          - s3:GetEncryptionConfiguration
+          - s3:GetInventoryConfiguration
+          - s3:GetBucketLogging
+          - s3:GetMetricsConfiguration
+          - s3:GetBucketNotification
+          - s3:GetBucketVersioning
+          - s3:GetReplicationConfiguration
+          - S3:GetBucketWebsite
+          - s3:GetBucketPublicAccessBlock
+          - s3:GetBucketObjectLockConfiguration
+          - s3:GetBucketTagging
+          - s3:GetBucketOwnershipControls
+          - s3:GetIntelligentTieringConfiguration
+          - s3:ListBucket
+        """
+        raise NotImplementedError
+
+    def delete(
+        self,
+        request: ResourceRequest[S3BucketProperties],
+    ) -> ProgressEvent[S3BucketProperties]:
+        """
+        Delete a resource
+
+        IAM permissions required:
+          - s3:DeleteBucket
+        """
+        model = request.desired_state
+        s3_client = request.aws_client_factory.s3
+
+        try:
+            s3_client.delete_bucket_policy(Bucket=model["BucketName"])
+        except s3_client.exceptions.ClientError:
```

Comment:
In original code of BaseLegacyProvider we were catching Exception. Instead of it, I went with more specific exception here.
Should I revert it to Exception?

Reply:
@bentsku Probably would know if the NotFound exception is considered a ClientError.

Reply:
why do we need to manually delete the bucket policy in the first place? Shouldn't it be deleted by deleting the bucket?

Reply:
I've removed explicit deletion of bucket policies and replaced exception with more general `ClientError`
Let me know If there is anything else I can do to improve the code.